app/src/main/java/me/martinez/sergio/daggertwobasearchitecture/activities/MainActivityComponentProd.java

```diff
@@ -0,0 +1,38 @@
+/*
+ *  Copyright (C) 2015 Sergio Martinez Open Source Project
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package me.martinez.sergio.daggertwobasearchitecture.activities;
+
+import javax.inject.Provider;
+
+import dagger.Subcomponent;
+import me.martinez.sergio.daggertwobasearchitecture.di.components.SectionComponent;
+import me.martinez.sergio.daggertwobasearchitecture.di.modules.ActivityModule;
+import me.martinez.sergio.daggertwobasearchitecture.di.anotations.scopes.PerActivity;
+import me.martinez.sergio.daggertwobasearchitecture.fragments.MainFragmentComponent;
+
+@PerActivity
+@Subcomponent(modules = {ActivityModule.class, MainActivityModule.class})
```

Comment:
You mean that before the relationship was `depends` and now is a subcomponent? I decided to migrate the whole thing to the new API, perhaps I misunderstood that part and not everything needs to be a subcomponent. 
For what I know, having it as sub-component is not the only way to make it get the corresponding bindings, but it would make sense if you want to define some sort of hierarchy. I think for Application-> Activity would make sense.